.github/workflows/linter-and-tests.yml

```diff
@@ -5,7 +5,6 @@ on:
     branches:
       - '**'
```

Comment:
Do we need to do this only on `main` branch because other branch will be merge from a PR ?

Reply:
Let's try this first and maybe limit the runs later.